treatment-service/models/storage.go

```diff
@@ -571,23 +573,21 @@ func NewLocalStorage(
 	clientOptions := []managementClient.ClientOption{}
 	if authzEnabled {
 		var googleClient *http.Client
-		// Init Google client for Authz. XP Server must be behind MLP Auth proxy, for authorization to work.
+		var err error
+		// Init Google client for Authz. When using a non-empty googleApplicationCredentialsEnvVar that contains a file
+		// path to a credentials file, the credentials file MUST contain a Google SERVICE ACCOUNT for authentication to
+		// work correctly
 		if filepath := os.Getenv(googleApplicationCredentialsEnvVar); filepath != "" {
-			data, err := os.ReadFile(filepath)
-			if err != nil {
-				return nil, err
-			}
-			creds, err := google.CredentialsFromJSON(context.Background(), data, GoogleOAuthScope)
-			if err != nil {
-				return nil, err
-			}
-			googleClient = oauth2.NewClient(context.Background(), creds.TokenSource)
+			googleClient, err = idtoken.NewClient(
+				context.Background(),
+				defaultCaraMLAudience,
+				idtoken.WithCredentialsFile(filepath),
+			)
```

Comment:
Will be replacing this block with the new method `InitGoogleClientFromCredentialsFile` introduced in https://github.com/gojek/mlp/pull/79 (after this gets approved and merged) which initialises a Google client a specified file path for both service AND user accounts.